# 11주차 회귀분석 연습문제 8장 R 풀이
> 11주차 회귀분석 연습문제 8장 R 풀이

- toc:true
- branch: master
- badges: true
- comments: true
- author: 김동준
- categories : ["R", "회귀분석"]

# 8.1)
> 어떤 철 제품을 생산하는 공장에서 제품의 인장강도 ($kg/cm^2$)가 온도와 시간에 어떤영향을 받는가를 분석하기 위하여 다음과 같은 자료를 얻었다.

|공정온도:$x_1$|시간(분):$x_2$|인장강도:y|
|:---:|:---:|:---:|
|175|15|120.4|
|210|18|112.5|
|192|20|95.4|
|250|28|162.3|
|245|25|160.2|
|226|21|131.5|
|260|32|157.6|
|230|25|158.4|
|205|24|149.6|
|185|19|130.4|

### 1)선형회귀모형 $y=\beta_0+\beta_1x_1+\beta_2x_2+\epsilon$ 이 성립한다고 가정하고 회귀모형을 추정하여라.

In [1]:
fac <- read.csv("8-1csv.csv")
fac

x1  x2 y_   
1  175 15 120.4
2  210 18 112.5
3  192 20  95.4
4  250 28 162.3
5  245 25 160.2
6  226 21 131.5
7  260 32 157.6
8  230 25 158.4
9  205 24 149.6
10 185 19 130.4

In [2]:
attach(fac)

In [5]:
lm81 <- lm(y_~x1+x2,data=fac)

In [6]:
coef(lm81)

(Intercept)          x1          x2 
 35.3358543   0.2047511   2.5506326

추정된 y는 $y=35.3359+ 0.2048x_1 + 2.5506x_2 $ 이다.

### 2) 추정된 회귀계수들에 대하여 $Var(b_0),Var(b_1),Var(b_2)$ 를 구하여라.

In [7]:
X <- matrix(c(1,1,1,1,1,1,1,1,1,1,array(fac$x1),array(fac$x2)),10,3)
beta <- matrix(c(35.3359,0.2048,2.5506),3,1)

In [8]:
XtX <- t(X) %*% X  #[X'X]
XtX

[,1] [,2]   [,3] 
[1,]   10   2178   227
[2,] 2178 481940 50621
[3,]  227  50621  5385

In [9]:
XtXr <- solve(XtX)  # [X'X]의 역함수
XtXr

[,1]        [,2]         [,3]        
[1,]  8.99974754 -0.065260222  0.234093777
[2,] -0.06526022  0.000637595 -0.003242642
[3,]  0.23409378 -0.003242642  0.020799716

$Var(b_0)$ : XtXr 의 대각 1번째 원소 $\times$  $\sigma^2$
> $8.9997 \sigma^2$

$Var(b_1)$ : XtXr 의 대각 2번째 원소 $\times$  $\sigma^2$
> $0.00064 \sigma^2$

$Var(b_2)$ : XtXr 의 대각 3번째 원소 $\times$  $\sigma^2$
> $0.0208 \sigma^2$

### 3) 추정된 회귀계수 $b_1$과 $b_2$의 의미는 무엇인가?

$b_1$ : $1^\circ C$ 온도 증가에 따른 인장강도의 변화

$b_2$ : $1$분 시간 흐름에 따른 인장강도의 변화

### 4) 회귀계수 $\beta_1$ 과 $\beta_2$의  95% 신뢰구간을 구하여라.

In [10]:
confint(lm81)

2.5 %       97.5 %    
(Intercept) -79.4362406 150.107949
x1           -0.7612855   1.170788
x2           -2.9669628   8.068228

$ -0.7612855 < \beta_1 < 1.170788 $ 

$ -2.9669628 < \beta_2 < 8.068228 $ 

### 5) 분산분석표를 작성하고 $H_0 : \beta_1 = \beta_2 = 0 $ 을 유의수준  0.05로 검정하여라.

In [11]:
anova(lm81)

Df Sum Sq    Mean Sq   F value   Pr(>F)    
x1        1  2747.5393 2747.5393 10.496050 0.01425777
x2        1   312.7796  312.7796  1.194869 0.31053404
Residuals 7  1832.3821  261.7689        NA         NA

분산분석표

||자유도|제곱합|제곱평균|F|
|:---:|:---:|:---:|:---:|:---:|
|model|2|3060|1530|6|
|error|7|1832|262||
|total|9|4892|||

유의수준 0.05 에서 F의 값보다 분산분석에서의 Fvalue(6)가 크므로 영가설을 기각하여

$\beta_1$ 과 $\beta_2$ 가 다르다는 것을 알 수 있다.

### 6) $x_1 = 200, x_2 = 20 $ 에서 $E(y)$ 의 95% 예측구간을 구하여라.

In [21]:
q11 <- data.frame(x1 = 200 , x2 = 20)

In [22]:
predict(lm81, q11, level=0.95 ,interval = "predic")

fit      lwr      upr     
1 127.2987 86.41531 168.1821

$\hat{y} \pm t \times \sqrt{x'(X'X)^{-1}x \bullet MSE} $

$\hat{y} = 127.3 , t = 2.365 , MSE = 262 , $

$ x'(X'X)^{-1}x : x = c(1, 200, 20) $ 으로 계산 $\to 0.142$ 

In [32]:
x_ <- matrix(c(1,200,20),3,1)
xt_ <- t(x_)
c_ <- (xt_) %*% (XtXr) %*% (x_)
c_

[,1]     
[1,] 0.1419628

In [33]:
127.3 + 2.365*sqrt(0.142*262)

[1] 141.7253

In [34]:
127.3 - 2.365*sqrt(0.142*262)

[1] 112.8747

따라서 예측구간은 112.9 ~ 141.7 이다.

### 7) 귀무가설 $H_0 : \beta_1 = \beta_2$, 대립가설 $H_1 : \beta_1 \neq \beta_2$ 를 유의수준 5%로 검정하여라.

In [37]:
q = matrix(c(0,1,-1),3,1)
b = matrix(c(35.3359,0.2048,2.5506),3,1)
qb = t(q) %*% b
qb

[,1]   
[1,] -2.3458

In [39]:
qb/sqrt(t(q) %*% XtXr %*% q *262) #검정 t통계량

[,1]      
[1,] -0.8672862

0.87 은 2.365보다 작으므로 영가설을 기각할 수 없다. 즉 $\beta_1 = \beta_2 $ 가 성립한다.

### 8) $x_1 =240 , x_2 =30 $ 에서 y의 95% 예측구간을 구하여라.

In [19]:
q10 <- data.frame(x1 = 240 , x2 = 30)

In [20]:
predict(lm81, q10, level=0.95 ,interval = "predic")

fit      lwr      upr     
1 160.9951 114.5839 207.4063

예측구간 : (114.5839 , 207.4063)의 사이에 y가 존재할 확률이 95%라는 뜻이다.

________________

# 8.10)
> 다음의 자료에 대하여

|$y$|$x_1$|$x_2$|
|---|---|---|
|39|2|4|
|42|2|4|
|51|3|4|
|48|3|4|
|53|2|6|
|49|2|6|
|61|3|6|
|60|3|6|

### 1) 중회귀모형 $y=\beta_0+\beta_1x_1+\beta_2x_2+\epsilon$를 가정하고 회귀선을 추정하여라.

In [40]:
Q810 <- read.csv("810.csv",head = T)
Q810

y_10 x1_10 x2_10
1 39   2     4    
2 42   2     4    
3 51   3     4    
4 48   3     4    
5 53   2     6    
6 49   2     6    
7 61   3     6    
8 60   3     6

In [41]:
attach(Q810)

In [42]:
lm810 <- lm(y_10~x1_10+x2_10)

In [43]:
coef(lm810)

(Intercept)       x1_10       x2_10 
      0.375       9.250       5.375

$\hat{y} = 0.375 + 9.25x_1 +5.375x_2 $

### 2) 독립변수 $x_1$만을 고려한 회귀모형 $y=\beta_0+\beta_1x_1+\epsilon$를 가정하고 회귀선을 구하여 분산분석표를 작성하여라.

In [44]:
lm810_1 <- lm(y_10 ~ x1_10)
coef(lm810_1)

(Intercept)       x1_10 
      27.25        9.25

$\hat{y} = 27.25 + 9.25x_1 $

ANOVA

In [45]:
anova(lm810_1)

Df Sum Sq  Mean Sq   F value  Pr(>F)    
x1_10     1  171.125 171.12500 4.127638 0.08846031
Residuals 6  248.750  41.45833       NA         NA

### 3) 독립변수 $x_2$만을 고려한 회귀모형 $y=\beta_0+\beta_2x_2+\epsilon$를 가정하고 회귀선을 구하여 분산분석표를 작성하여라.

In [46]:
lm810_2 <- lm(y_10 ~ x2_10)
coef(lm810_2)

(Intercept)       x2_10 
     23.500       5.375

$\hat{y} = 23.5 +5.375x_2 $

ANOVA

In [47]:
anova(lm810_2)

Df Sum Sq  Mean Sq   F value Pr(>F)    
x2_10     1  231.125 231.12500 7.34702 0.03508095
Residuals 6  188.750  31.45833      NA         NA

### 4) 단위길이법에 의하여 표준화된 회귀계수를 구하여 회귀선을 추정하여라.

-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-

$y=0.6384x_1 + 0.7419x_2$ 을 추정할 수 있다.

### 5) 표준화된 회귀모형에 의하여 분산분석표를 작성하여라.

-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-

### 6) 4)에서 계산된 회귀계수들의 상관행렬을 계산하고 상관이 존재하는지를 설명하여라.

-
-
-
-
-
-
-

_________________________________

# 8.15)
> 다음과 같은 자료에 대하여 중회귀모형 $y=\beta_0+\beta_1x_1+\beta_2x_2+\epsilon$를 가정하고 아래의 질문에 답하여라.

|$y$|$x_1$|$x_2$|
|---|---|---|
|11|-5|5|
|11|4|4|
|8|-1|1|
|2|2|-3|
|5|2|-2|
|5|3|-2|
|4|3|-3|

### 1) 분산분석표를 작성하고 $H_0 : \beta_1 = \beta_2 = 0 $ 을 $\alpha = 0.05 $로 검정하여라.

In [6]:
y15 <- c(11,11,8,2,5,5,4)
x1_15 <- c(-5,4,-1,2,2,3,3)
x2_15 <- c(5,4,1,-3,-2,-2,-3)
lm815 <- lm(y15 ~ x1_15+x2_15)
coef(lm815)

(Intercept)       x1_15       x2_15 
  6.4331053   0.1210329   1.0770022

$y=6.43310529293761 + 0.121032868679595x_1 + 1.07700221182038x_2$

In [7]:
anova(lm815)

Df Sum Sq    Mean Sq    F value  Pr(>F)     
x1_15     1  15.879334 15.8793343 20.48100 0.010612931
x2_15     1  54.733671 54.7336707 70.59492 0.001098159
Residuals 4   3.101281  0.7753202       NA          NA

Fvalue 가 6.94 보다 크므로 영가설을 기각하여 $\beta_1$ 과 $\beta_2$ 가 다르다는 것을 알 수 있다.

### 2) $R(\beta_0), R(\beta_1,\beta_2|\beta_0),R(\beta_1|\beta_0),R(\beta_2|\beta_0)$ 을 각각 구하여라.

$R(\beta_1,\beta_2|\beta_0)$ = 15.879334

In [16]:
anova(lm815)

Df Sum Sq    Mean Sq    F value  Pr(>F)     
x1_15     1  15.879334 15.8793343 20.48100 0.010612931
x2_15     1  54.733671 54.7336707 70.59492 0.001098159
Residuals 4   3.101281  0.7753202       NA          NA

$R(\beta_1|\beta_0) $=15.87933

In [17]:
lm815a = lm(y15~x1_15)
anova(lm815a)

Df Sum Sq   Mean Sq  F value  Pr(>F)   
x1_15     1  15.87933 15.87933 1.372815 0.2941169
Residuals 5  57.83495 11.56699       NA        NA

$R(\beta_2|\beta_0)$ =70.01471

In [10]:
lm815b = lm(y15~x2_15)
anova(lm815b)

Df Sum Sq   Mean Sq   F value  Pr(>F)      
x2_15     1  70.01471 70.014706 94.62521 0.0001951318
Residuals 5   3.69958  0.739916       NA           NA

$R(\beta_0)$ = $ n(\bar{y})^2$ =302.285714285714

In [15]:
7*mean(y15)^2

[1] 302.2857

### 3)  $H_0 : \beta_1 = 0 $ 을 부분 F검정으로 가설검정하여라.  $\alpha = 0.05 $

In [19]:
anova(lm815b,lm815)

Res.Df RSS      Df Sum of Sq F         Pr(>F)   
1 5      3.699580 NA       NA         NA        NA
2 4      3.101281  1 0.598299  0.7716799 0.4292997

Fvalue 는 0.7717 이고, 7.71 보다 작으므로 영가설을 기각하지 않는다. 즉, $\beta_1 = 0$ 이 성립한다.  

### 4)  $H_0 : \beta_2 = 0 $ 을 부분 F검정으로 가설검정하여라.  $\alpha = 0.05,\alpha = 0.01 $

In [20]:
anova(lm815a,lm815)

Res.Df RSS       Df Sum of Sq F        Pr(>F)     
1 5      57.834951 NA       NA        NA          NA
2 4       3.101281  1 54.73367  70.59492 0.001098159

Fvalue = 70.595로 측정된다.

유의수준 0.05일때 F값 7.71 보다 F값이 크며,  유의수준 0.01일때 F값인 21.2 보다도 F값이 크다.

즉, 영가설은 어느 유의수준에서도 기각되어 $\beta_2$ 가 0이 아니라는 것으로 볼 수 있다.

____

# 8.16)
> 중회귀모형 $y= X\beta + \epsilon, \epsilon ~ N(0,I)$ 에서 $X$는 $n \times (k+1)$ 이다.

### 1) $Q_1 = y'[I-X(X'X)^{-1}X']y$ 의 분포는?

-
-
-
-
-
-
-
-
-

### 2) $Q_2 = y'X(X'X)^{-1}X'y$ 의 분포는?

-
-
-
-
-
-
-
-
-

### 3) $Q_1$과$Q_2$가 서로 독립임을 보여라.

-
-
-
-
-
-
-
-
-

### 4) $\beta = 0 $ 이라 할 때

$\frac{Q_2/(k+1)}{Q_1/(n-k-1)}$ 의 분포는?

-
-
-
-
-
-
-
-
-